packages/rum/src/domain/record/record.ts

```diff
@@ -42,28 +48,53 @@ export function record(options: RecordOptions): RecordAPI {
     throw new Error('emit function is required')
   }
 
-  const mutationController = new MutationController()
+  const shadowDomCallBacks = new Map<ShadowRoot, ShadowDomCallBacks>()
   const elementsScrollPositions = createElementsScrollPositions()
 
+  const shadowDomRemovedCallback = (shadowRoot: ShadowRoot) => {
+    const entry = shadowDomCallBacks.get(shadowRoot)
+    if (!entry) {
+      addTelemetryDebug('no shadow root in map')
+      return
+    }
+    entry.stop()
+    shadowDomCallBacks.delete(shadowRoot)
+  }
+
   const mutationCb = (mutation: BrowserMutationPayload) => {
     emit(assembleIncrementalSnapshot<BrowserMutationData>(IncrementalSource.Mutation, mutation))
   }
   const inputCb: InputCallback = (s) => emit(assembleIncrementalSnapshot<InputData>(IncrementalSource.Input, s))
-
   const shadowDomCreatedCallback = (shadowRoot: ShadowRoot) => {
-    startMutationObserver(mutationController, mutationCb, options.configuration, shadowDomCreatedCallback, shadowRoot)
+    const shadowDomMutationController = new MutationController()
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** drop mutation controller (seen IRL)